File: java/java_console/src/main/java/com/brasilburger/app/model/Complement.java
package com.brasilburger.app.model;

import com.brasilburger.app.enums.TypeComplement;

public class Complement {
    private int id;
    private String nom;
    private double prix;
    private String image;
    private TypeComplement typeComplement;

    public Complement(int id, String nom, double prix, String image, TypeComplement typeComplement) {
        this.id = id;
        this.nom = nom;
        this.prix = prix;
        this.image = image;
        this.typeComplement = typeComplement;
    }

    // Getters
    public int getId() {
        return id;
    }

    public String getNom() {
        return nom;
    }

    public double getPrix() {
        return prix;
    }

    public String getImage() {
        return image;
    }

    public TypeComplement getTypeComplement() {
        return typeComplement;
    }

    // Compatibility accessor
    public TypeComplement getType() {
        return typeComplement;
    }

    @Override
    public String toString() {
        return id + "] Complément (" + typeComplement + "): " + nom + " - " + prix + " CFA";
    }
}

File: src/Entity/Complement.php
<?php
namespace App\Entity;

use App\Repository\ComplementRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ComplementRepository::class)]
#[ORM\Table(name: 'COMPLEMENTS')]
class Complement
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: 'IDENTITY')]
    #[ORM\Column(name: 'id_complement')]
    private ?int $id = null;

    #[ORM\Column(length: 100)]
    private ?string $nom = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prix = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $image = null;

    #[ORM\Column(name: 'type_complement', length: 50)]
    private ?string $type = null;

    #[ORM\Column(name: 'est_archive', type: 'boolean', options: ['default' => false])]
    private bool $isArchived = false;

    
    public function getId(): ?int { return $this->id; }
    public function getNom(): ?string { return $this->nom; }
    public function setNom(string $nom): self { $this->nom = $nom; return $this; }
    public function getPrix(): ?string { return $this->prix; }
    public function setPrix(string $prix): self { $this->prix = $prix; return $this; }
    public function getType(): ?string { return $this->type; }
    public function setType(string $type): self { $this->type = $type; return $this; }
    public function getImage(): ?string { return $this->image; }
    public function setImage(?string $image): self { $this->image = $image; return $this; }
    public function isArchived(): bool { return $this->isArchived; }
    public function setIsArchived(bool $isArchived): self { $this->isArchived = $isArchived; return $this; }
}